tensordict/tensordict.py

```diff
@@ -1524,6 +1555,19 @@ def __len__(self) -> int:
         """Returns the length of first dimension, if there is, otherwise 0."""
         return self.shape[0] if self.batch_dims else 0
 
+    def __contains__(self, key):
+        # by default a Mapping will implement __contains__ by calling __getitem__ and
```

Comment:
probably this is out of scope of this PR but why not fall back to `return key in self.keys()` in this case? Membership check is a very common op and users would expect this to work.

Reply:
This is an open discussion.
Here's the summary of our thoughts, correct me if I'm missing something @tcbegley 

The default behaviour of `__contains__` is to try `__getitem__` and return False if an error is thrown. For the lazy tensordicts (views and stacks) getitem is expensive, so we don't want to do that.
- we try as much as possible to avoid behaviours that are not well defined and/or duplicated (there should not be more than one simple way of doing something). It's a general policy for pytorch. With a stack of tensordicts you might say "I want to know if this tensordict is in the stack" and use `in` for that. If we enable that it would be strange to use `in` timely look up for keys. I also think a good error message goes a long way.
Happy to reconsider though! I'm not 100% convinced by my own arguments :)

Reply:
Regarding `The default behaviour of __contains__ ...` I understand the rationale behind not supporting lazy tds so it is fine to raise an error for that case.

Regarding `behaviours that are not well defined and/or duplicated`, I think user experience triumps this. Even PT added numpy compatibility (eg supporting both dims and axis). This is how the user experience would look right now (feel free to correct me if I am wrong):

1. User tries `key in td`.
2. Runs into an error.
3. Calls key in td.keys()`

1 is surprising behavior, more so given that all python containers support "in". 3 is confusing because python programmers are conditioned to prefer `key in _dict` over `key in _dict.keys()` as the former is O(1) while the latter is O(n)